packages/uniforms-bootstrap4/src/gridClassName.js

```diff
@@ -38,7 +39,10 @@ export default function gridClassName (grid, side) {
 
     // Example: {xs: 6, sm: 4, md: 3}
     if (typeof grid === 'object') {
-        return Object.keys(grid).map(size => gridClassNamePart(size, grid[size], side)).join(' ');
+        const completeGrid = {xs: grid.sm || grid.md || grid.lg || grid.xl, ...grid};
```

Comment:
Move it into an `if`. This way we'd make a lot of unnecessary copies.

Reply:
Do you prefer
```
        if (!grid.xs) {
            const completeGrid = {xs: grid.sm || grid.md || grid.lg || grid.xl, ...grid};
            return Object.keys(completeGrid || grid)
                .sort(compareSizeClass)
                .map(size => gridClassNamePart(size, completeGrid[size], side)).join(' ');
        }

        return Object.keys(grid)
            .sort(compareSizeClass)
            .map(size => gridClassNamePart(size, grid[size], side)).join(' ');
```
or
```
const completeGrid = grid.xs
            ? grid
            : {xs: grid.sm || grid.md || grid.lg || grid.xl, ...grid};

        return Object.keys(completeGrid)
            .sort(compareSizeClass)
            .map(size => gridClassNamePart(size, completeGrid[size], side)).join(' ');
```

Reply:
I think our ESLint allows overriding the argument. If so, then

```js
if (!grid.xs) {
    grid = {xs: grid.sm || grid.md || grid.lg || grid.xl, ...grid};
}
```

If not, then the second one.